src/slashcommands/Nsfw.ts

```diff
@@ -47,18 +49,13 @@ export class Nsfw implements SlashCommand {
 		bot: Bot,
 		interaction: CommandInteraction<CacheType>
 	): Promise<void> {
-		let member = interaction.member as GuildMember;
-		if (!(interaction.channel instanceof TextChannel)) {
-			interaction.reply('Command must be used in a server');
-			return;
-		}
-		if (!member.permissionsIn(interaction.channel!).has('ADMINISTRATOR')) {
-			interaction.reply({
+		//check if the user is a Manager or Admin
+		if (!(await managerCheck(interaction.guild!, interaction.user))) {
```

Comment:
Since you're copy-pasting this code a bunch, I would condense this whole block into the managerCheck function.
Just have it do the replying and everything in the function, then do a return check.
```
if (!(await managerCheck(interaction))) return; 
```

Reply:
Or even better, create an optional boolean variable on the SlashCommand interface to set a command to require being a manager, then do the check in InteractionCreate